src/js/view/controls/components/menu/settings-menu.js

```diff
@@ -79,18 +81,26 @@ class SettingsMenu extends Menu {
     onAudioTracks(model, audioTracks) {
         this.setupMenu(
             'audioTracks', 
+            this.localization.audioTracks,
             audioTracks, 
             (index) => this.api.setCurrentAudioTrack(index), 
             model.get('currentAudioTrack')
         );
     }
+    onAudioTrackIndex(model, trackIndex) {
+        if (!this.children.audioTracks) {
+            return;
+        }
+        this.children.audioTracks.items[trackIndex].activate();
```

Comment:
are we sure this will always have a value? also how does the highlight currently work without this logic (when selected an item, i know this bug fix is for the first default selection). i worry if we're spreading our logic for highlighting throughout both modules even though this shells the call out to the submenu. some food for thought

Reply:
Ah, just saw this. It appears to always have a value. From ProgramController:
```
if (currentTrack > -1 && tracks.length > 0 && currentTrack < tracks.length) {
  mediaModel.set('currentAudioTrack', parseInt(currentTrack));
}
```

All listeners for external selection like `onAudioTrackIndex` are in settings-menu.js. There is a function that adds listeners to menu items for click, where it selects it and activates it if it isn't already active. Also the base activate function checks if active before activating as well.

Reply:
i guess my question is more does this logic make more sense in the child menu? is that possible to do or does the generic factory make that more trouble than it's worth? just feels a little strange to have a listener in the parent that then shells the calls out to the child

Reply:
Having all of the data-driven aspects of state under a single container makes it trivial to remove event listeners on destroy, and to debug if there are issues. Also, since the event lifecycle isn't normalized between various events (for example change:captions doesn't fire if the new array is empty, other change events do). Settings menu is our main menu, so it makes sense to treat it as our state container, with the data returned from those model listeners as the source of truth.